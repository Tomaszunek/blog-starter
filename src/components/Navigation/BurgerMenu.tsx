import * as React from 'react';
import Links from './navLinks';

export default class BurgerMenu extends React.Component<IProps, IState> {

    public constructor(props: IProps) {
        super(props);
        this.state = {
            showBurgerMenu: false
        }
    }
      
  public render() {
    const { showBurgerMenu } = this.state;
    const style = {
        display: (showBurgerMenu ? " block" : "none")
    }
    return (
        <div className={"burgerMenu" + (showBurgerMenu ? " change" : "")} onClick={(this.toggleBurger)}>
            <div className="bar1"/>
            <div className="bar2"/>
            <div className="bar3"/>
            <div className="burgerLinksWraper" style={style}>
                <Links/>
            </div>            
        </div>
    );
  }

  private toggleBurger = () => {
    this.setState({
        showBurgerMenu: !this.state.showBurgerMenu
    })
  };  
}

interface IProps {
    someVrbl?: any
}
interface IState {
    showBurgerMenu: boolean
}
  