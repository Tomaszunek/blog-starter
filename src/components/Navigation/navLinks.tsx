import * as React from 'react';
import { NavLink }  from 'react-router-dom';

const Links: React.SFC = () => {
 return (
    <ul>
        {/* <li className="login">
            <NavLink to="/login">login</NavLink>
        </li>
        <li>
            <NavLink to="/articles/technology">technology</NavLink>
        </li>
        <li>
            <NavLink to="/articles/psychology">psychology</NavLink>
        </li>
        <li>
            <NavLink to="/articles/motivation">motivation</NavLink>
        </li>
        <li>
            <NavLink to="/articles/spiritual">spiritual</NavLink>
        </li>
        <li>
            <NavLink to="/products">store</NavLink>
        </li> */}

        <li className="login">
            <NavLink to="/login">login</NavLink>
        </li>
        <li>
            <NavLink to="/articles/news">news</NavLink>
        </li>
        <li>
            <NavLink to="/projects">projects</NavLink>
        </li>
        <li>
            <NavLink to="/abilities">abilities</NavLink>
        </li>
        <li>
            <NavLink to="/products">store</NavLink>
        </li>
        <li>
            <NavLink to="/about">about</NavLink>
        </li>     
    </ul>
 )
}

export default Links;