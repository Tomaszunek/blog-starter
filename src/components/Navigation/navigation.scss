@import 'src/styles/variables.scss';

nav {
    background-color: $background-alpha;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding-left: 8%;
    padding-right: 8%;
    padding-top: 5px;
    padding-bottom: 5px;
    border-bottom: 1px solid $font-c;
    font-size: 18px;
    color: $font-white;
    position: fixed;
    width: 84%;
    top: 0px;
    left: 0px;
    z-index: 9999;

    &.scrolled {
        background-color: $background;
        color: $font-c;
        padding-top: 0;
        padding-bottom: 0;        
        

        a {
            color: $font-c;

            &.logo {
                font-size: 20px;
                display: flex;
                background: url($logo-black); 
                background-repeat: no-repeat;
                background-size: 22%;
                background-position: -1px 5px;
                padding: 18px 48px;
            }
    
            &:hover {
                color: $font-white;
            }
        }

        ul {
            li {
                border-right: 1px solid $font-c;
                
                a {
                    &:hover {
                        padding: 19px 14px;
                    }
                }
                
            }
        }
    }

    a {
        text-decoration: none;
        color: $font-white;

        &.logo {
            font-size: 20px;
            display: flex;
            background: url($logo-white); 
            background-repeat: no-repeat;
            background-size: 22%;
            background-position: -1px 5px;
            padding: 18px 49px;
        }

        &:hover {
            color: $font-white;
        }
    }

    

    ul {        
        display: flex;
        list-style: none;        

        li {            
            border-right: 1px solid $font-white;
            &:last-child {
                border-right: 0px;                
            }

            a {
                padding: 5px 19px;                
                text-transform: uppercase;

                &:hover {
                    background-color: $theme-c;
                    color: $font-white;
                    padding: 24px 14px;
                    margin: 0 5px;           
                }        
            }            

            &.login {
                display: none;
            }
        }
    }

    .login {
        text-transform: uppercase;        
    }

    .burgerMenu {
        display: none;
        cursor: pointer;
        position: relative;

        .bar1, .bar2, .bar3 {
            width: 30px;
            height: 5px;
            background-color: $theme-c;
            margin: 6px 0;
            transition: 0.4s;
        }
        
        &.change .bar1 {
            -webkit-transform: rotate(-45deg) translate(-9px, 6px);
            transform: rotate(-45deg) translate(-9px, 6px);
        }
        
        &.change .bar2 {opacity: 0;}
        
        &.change .bar3 {
            -webkit-transform: rotate(45deg) translate(-8px, -8px);
            transform: rotate(45deg) translate(-8px, -8px);
        }
        .burgerLinksWraper {
            position: absolute;
            left: -160px;
            display: none;
            border: 1px solid $font-white;
            background-color: $theme-c;
            ul {                
                display: block;
                padding: 0px 20px;
                font-size: 20px;

                li {                    
                    border-right: 0px;
                    border-bottom: 1px solid $font-white;
                    text-align: left;
                    margin: 5px 0px;
                    a {
                        padding: 6px 7px;
                        margin: 4px 0px;                     
                        
                        &:hover {
                            background-color: $font-white;
                            color: $font-black;
                            padding: 0px 7px;
                            margin: 0;           
                        }  
                    }
        
                    
                    &.login {
                        display: block;
                    }                 
                }
            }        
        }        
    }   

    @media only screen and (min-width: $sm) and (max-width: $lg) {        
        font-size: 16px;       
    }
    @media only screen and (min-width: $xs) and (max-width: $sm) {        
        font-size: 14px;

        .logo {
            font-size: 16px;
        }

        ul {
            display: none;
        }

        .login {
            display: none;
        }

        .burgerMenu {
            display: inline-block;            
        }

        
    }
    @media only screen and (max-width: $xs) {        
        font-size: 12px;

        .logo {
            font-size: 14px;
        }

        ul {
            display: none;            
        }

        .login {
            display: none;
        }

        .burgerMenu {
            display: inline-block;            
        }
    }   
}