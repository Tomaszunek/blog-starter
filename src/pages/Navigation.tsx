import * as React from 'react';

export default class Navigation extends React.Component {    
  public render() {
    return (
      <nav className="Footer">
        abcd
      </nav>
    );
  }
}


