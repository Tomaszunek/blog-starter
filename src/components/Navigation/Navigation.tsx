import * as React from 'react';
import Links from './navLinks';

import './navigation.scss';
import BurgerMenu from './BurgerMenu';

export default class Navigation extends React.Component {    
  public render() {
    return (
      <nav className="navigation">
        <div className="logo">
          Prohuman
        </div>
        <Links/>
        <div className="login">
          Login
        </div>
        <BurgerMenu/>
      </nav>
    );
  }
}