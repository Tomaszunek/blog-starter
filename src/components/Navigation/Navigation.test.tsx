import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import Navigation from './Navigation'
import { BrowserRouter as Router } from 'react-router-dom';

describe('Burger Menu', () => {
  describe('Component should render right', () => {
    let app:any;             
    beforeEach(() => {
      app = mount( 
      <Router>
        <Navigation />
      </Router>);
    })
    it('render correctly snapshot component', () => {
      const tree = renderer
      .create( 
      <Router>
        <Navigation />
      </Router>)
      .toJSON();
      expect(tree).toMatchSnapshot();
    })    
    it('should contain nav containar', () => {      
      expect(app.find('nav').length).toEqual(1)        
    })
    it('should contain elements', () => {
      expect(app.find('.burgerMenu').length).toEqual(1);
      expect(app.find('.burgerLinksWraper').length).toEqual(1);      
    })    
  });
}); 