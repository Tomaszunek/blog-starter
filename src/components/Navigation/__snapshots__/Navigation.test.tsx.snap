// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Burger Menu Component should render right render correctly snapshot component 1`] = `
<nav
  className=""
>
  <a
    aria-current="page"
    className="logo active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <span>
      prohuman 
      games
    </span>
  </a>
  <ul>
    <li
      className="login"
    >
      <a
        aria-current={null}
        className={undefined}
        href="/login"
        onClick={[Function]}
        style={undefined}
      >
        login
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className={undefined}
        href="/articles/news"
        onClick={[Function]}
        style={undefined}
      >
        news
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className={undefined}
        href="/projects"
        onClick={[Function]}
        style={undefined}
      >
        projects
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className={undefined}
        href="/abilities"
        onClick={[Function]}
        style={undefined}
      >
        abilities
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className={undefined}
        href="/products"
        onClick={[Function]}
        style={undefined}
      >
        store
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className={undefined}
        href="/about"
        onClick={[Function]}
        style={undefined}
      >
        about
      </a>
    </li>
  </ul>
  <div
    className="login"
  >
    <a
      aria-current={null}
      className={undefined}
      href="/login"
      onClick={[Function]}
      style={undefined}
    >
      login
    </a>
  </div>
  <div
    className="burgerMenu"
    onClick={[Function]}
  >
    <div
      className="bar1"
    />
    <div
      className="bar2"
    />
    <div
      className="bar3"
    />
    <div
      className="burgerLinksWraper"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <ul>
        <li
          className="login"
        >
          <a
            aria-current={null}
            className={undefined}
            href="/login"
            onClick={[Function]}
            style={undefined}
          >
            login
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className={undefined}
            href="/articles/news"
            onClick={[Function]}
            style={undefined}
          >
            news
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className={undefined}
            href="/projects"
            onClick={[Function]}
            style={undefined}
          >
            projects
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className={undefined}
            href="/abilities"
            onClick={[Function]}
            style={undefined}
          >
            abilities
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className={undefined}
            href="/products"
            onClick={[Function]}
            style={undefined}
          >
            store
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className={undefined}
            href="/about"
            onClick={[Function]}
            style={undefined}
          >
            about
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
